### PR TITLE
remove meta tags

### PR DESCRIPTION
**Pull request by jazzsequence**
Removes the hm-metatags plugin and replaces it with integration with the Yoast SEO social/metadata options.

### CODE REVIEW

File: inc/namespace.php [hunk 1]
```diff
@@ -125,88 +132,142 @@ function remove_yoast_submenu_page() {
...
+	// If the fallback image ID was not configured, but we do have a fallback image, get the ID from the URL.
+	if ( empty( $options['og_default_image_id'] ) && $options['og_default_image'] ) {
+		$options['og_default_image_id'] = get_image_id_from_url( $config['fallback-image'] );
+	}
+
+	// These options are only used as fallbacks from the default Home and Front Page SEO options, and possibly not even then.
+	$options['og_frontpage_title'] = $config['opengraph-fallback']['og-frontpage-title'] ?? '';
+	$options['og_frontpage_desc'] = $config['opengraph-fallback']['og-frontpage-desc'] ?? '';
+	$options['og_frontpage_image'] = $config['opengraph-fallback']['og-frontpage-image'] ?? '';
+	$options['og_frontpage_image_id'] = $config['opengraph-fallback']['og-frontpage-image-id'] ?? '';
```
Comment (roborourke) [inline inc/namespace.php:201]: Should these fall back to the default image if not set? Not sure if Yoast does that internally or not

File: inc/namespace.php [hunk 2]
```diff
@@ -125,88 +132,142 @@ function remove_yoast_submenu_page() {
...
+ * Compares the SEO metadata options saved in the config file with the default values. If anything has been changed, returns true.
  *
- * @return array Twitter metadata with image using Tachyon URL, if any.
+ * @return bool True if metadata values have been saved.
  */
-function metadata_img_as_tachyon_twitter( array $meta ) : array {
-	return metadata_img_as_tachyon( $meta, [
-		'resize' => '1200,600', // crop.
-	] );
+function should_override_metadata_options() : bool {
```
Comment (roborourke) [inline inc/namespace.php:218]: So as soon as one option is set in config we switch off the admin page? Pity we can't do it more granularly :/

File: inc/opengraph/class-altis-opengraph-author-presenter.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+namespace Altis\SEO\Opengraph;
+
+use Yoast\WP\SEO\Presenters\Abstract_Indexable_Tag_Presenter;
+
+/**
+ * Adds a custom Opengraph Article Author meta tag.
+ */
+class Altis_Opengraph_Author_Presenter extends Abstract_Indexable_Tag_Presenter {
```
Comment (roborourke) [inline inc/opengraph/class-altis-opengraph-author-presenter.php:15]: We're already in both `Altis` and `Opengraph` namespaces so I think just `Author_Presenter` would be fine and remove the redundancy. Same for the other classes.

File: inc/namespace.php [hunk 4]
```diff
@@ -125,88 +132,142 @@ function remove_yoast_submenu_page() {
...
-	// Stop - no image for metadata.
-	if ( ! isset( $meta['image'] ) ) {
-		return $meta;
-	}
+function get_image_id_from_url( string $url ) {
+	global $wpdb;
+	$url = preg_replace( '/-\d+x\d+(?=\.(jpg|jpeg|png|gif)$)/i', '', $url );
+
+	$image = $wpdb->get_col( $wpdb->prepare(
+		"SELECT ID FROM $wpdb->posts WHERE guid=%s;",
```
Comment (roborourke) [inline inc/namespace.php:262]: This won't be multisite aware and could end up running twice on every page load, the guid isn't a reliable indicator of the attachment URL and additionally this query isn't cached.

There's a built in WP function `attachment_url_to_postid()` too but that's also not multisite aware. Can we avoid doing this at all? Will Yoast freak out if it doesn't know the ID?

File: inc/namespace.php [hunk 5]
```diff
@@ -230,6 +291,22 @@ function override_yoast_seo_options( $options ) : ?array {
...
+ */
+function social_options_overridden_notice() {
+	$screen = get_current_screen();
+
+	if ( $screen->base !== 'toplevel_page_wpseo_dashboard' ) {
+		return;
+	}
+
+	$classes = 'notice notice-warning is-dismissable';
+	$message = __( 'Social metadata has been saved in the Altis configuration file. Use the Altis configuration file to make changes to the Social SEO settings.', 'altis-seo' );
```
Comment (roborourke) [inline inc/namespace.php:305]: Could maybe provide a link to the docs on this 🤔  Would need to extract the version and stuff to get the right link though so maybe not worth it right now.

Also maybe reword the language around "metadata saved in the config file", perhaps just "set", minor thing it just made me pause. Because the call to action "Use the config file to make changes" is written there that might be confusing for non developers or people using Altis who aren't aware of the technical components.

I would be tempted to just drop this notice altogether, or remove the second sentence or make it clear who's responsibility it would be to make changes. Maybe like:

```suggestion
	$message = __( 'Social media profile data has been set in the Altis configuration file and cannot be modified here.', 'altis-seo' );
```

**Review by roborourke — CHANGES_REQUESTED** [review]
My main concern is the attachment URL to post ID function, if we can we should get rid of that. I think if you have a URL already you shouldn't need an ID and if you have an ID you can derive the URL easily.

File: docs/README.md [hunk 6]
```diff
@@ -18,27 +18,27 @@ The following JSON is the default configuration for the module and can be overri
 					"metadata": {
 						"opengraph": true,
 						"twitter": true,
-						"json-ld": true,
-						"fallback-image": "",
+						"fallback-image": false,
+						"fallback-image-id": "",
```
[the remark this answers is not in the document]
Comment (jazzsequence) [reply docs/README.md:22]: Good call, we can probably pull it out.

File: docs/README.md [hunk 7]
```diff
@@ -18,27 +18,27 @@ The following JSON is the default configuration for the module and can be overri
...
 							"linkedin": "",
 							"myspace": "",
 							"pinterest": "",
-							"soundcloud": "",
-							"tumblr": "",
+							"youtube": "",
+							"wikipedia": ""
 						},
+						"opengraph-fallback": {
+							"og-frontpage-title": "",
```
[the remark this answers is not in the document]
Comment (jazzsequence) [reply docs/README.md:35]: I'm actually wondering if these settings are used or even needed. They feel like legacy options. In my testing, I couldn't find anything that actually used them. What's your thought about including the config options in the `load.php` file but just not documenting them? I.e. it's there for backpat but you probably wouldn't need it? Too much cruft? Should we just remove entirely if it doesn't look like something that's in use?

File: inc/namespace.php (hunk 2 above)
Comment (jazzsequence) [reply inc/namespace.php:218]: Yeah, all of these settings are stored as a single serialized option value, so as soon as you use `pre_option_wpseo_social`, you're overriding anything that might've been saved in the database. In fact, it actually clears out the database options if you switch, which is why I thought removing the admin page was better. You can't make changes that save if you're using the config at all and when you switch, everything that was saved in the config is not stored in the database so it's all empty.

File: inc/namespace.php (hunk 4 above)
Comment (jazzsequence) [reply inc/namespace.php:262]: I will test without. I don't _think_ there's any risk but I know that when you save the settings normally on the settings page, it pulls from the media library specifically and therefore stores the ID.

File: docs/README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (jazzsequence) [reply docs/README.md:35]: I'll leave them so they can be used but drop the prefix. These look like they are mostly for sites where the homepage displays blog posts rather than a static front page.

File: inc/namespace.php (hunk 1 above)
Comment (jazzsequence) [reply inc/namespace.php]: Apparently not. 👍

File: inc/namespace.php (hunk 5 above)
Comment (jazzsequence) [reply inc/namespace.php:264]: Updating this a bit to only display once in addition to changing the language.

**Review by roborourke — APPROVED** [review]
Nice work
